server/src/test/java/org/opensearch/ratelimitting/admissioncontrol/AdmissionControlSingleNodeTests.java

```diff
@@ -0,0 +1,203 @@
+        assertEquals(1, (long) acStats.getRejectionCount().get(AdmissionControlActionType.INDEXING.getType()));
+        client().admin().indices().prepareRefresh("index").get();
+
+        // verify search request hits 429
+        SearchRequest searchRequest = new SearchRequest("index");
+        try {
+            client().search(searchRequest).actionGet();
+        } catch (Exception e) {
+            assertTrue(((SearchPhaseExecutionException) e).getDetailedMessage().contains("OpenSearchRejectedExecutionException"));
+        }
+        acStats = admissionControlService.stats().getAdmissionControllerStatsList().get(0);
+        assertEquals(1, (long) acStats.getRejectionCount().get(AdmissionControlActionType.SEARCH.getType()));
+    }
+
+    public void testAdmissionControlRejectionMonitorOnlyMode() throws Exception {
+        assertBusy(() -> assertEquals(1, getInstanceFromNode(ResourceUsageCollectorService.class).getAllNodeStatistics().size()));
+        // Verify that cluster state is updated
+        ActionFuture<ClusterStateResponse> future2 = client().admin().cluster().state(new ClusterStateRequest());
+        assertThat(future2.isDone(), is(true));
+
+        ClusterUpdateSettingsRequest updateSettingsRequest = new ClusterUpdateSettingsRequest();
+        updateSettingsRequest.transientSettings(
+            Settings.builder()
+                .put(super.nodeSettings())
+                .put(ADMISSION_CONTROL_TRANSPORT_LAYER_MODE.getKey(), AdmissionControlMode.MONITOR.getMode())
+        );
+        assertAcked(client().admin().cluster().updateSettings(updateSettingsRequest).actionGet());
+
+        BulkRequestBuilder bulk = client().prepareBulk();
+        for (int i = 0; i < 3; i++) {
+            bulk.add(client().prepareIndex("index").setSource("foo", "bar " + i));
+        }
+        // verify bulk request success but admission control having rejections stats
+        BulkResponse res = client().bulk(bulk.request()).actionGet();
+        assertFalse(res.hasFailures());
+        AdmissionControlService admissionControlService = getInstanceFromNode(AdmissionControlService.class);
+        AdmissionControllerStats acStats = admissionControlService.stats().getAdmissionControllerStatsList().get(0);
+        assertEquals(1, (long) acStats.getRejectionCount().get(AdmissionControlActionType.INDEXING.getType()));
+        client().admin().indices().prepareRefresh("index").get();
+
+        // verify search request success but admission control having rejections stats
+        SearchRequest searchRequest = new SearchRequest("index");
+        SearchResponse searchResponse = client().search(searchRequest).actionGet();
+        assertEquals(3, searchResponse.getHits().getHits().length);
+        acStats = admissionControlService.stats().getAdmissionControllerStatsList().get(0);
+        assertEquals(1, (long) acStats.getRejectionCount().get(AdmissionControlActionType.SEARCH.getType()));
```

Comment:
If this is monitor mode, why should we expect rejections?

Reply:
In monitor only mode we will not reject the requests but we will track the rejections and update stats.